lxc/image.go

```diff
@@ -286,3 +307,90 @@ func (c *imageCmd) run(config *lxd.Config, args []string) error {
 		return fmt.Errorf(gettext.Gettext("Unknown image command %s"), args[0])
 	}
 }
+
+func dereferenceAlias(d *lxd.Client, inName string) string {
+	imageList, err := d.ListImages()
+	if err != nil {
+		return ""
+	}
+	inLen := len(inName)
+	for _, urln := range imageList {
+		prefix := "/1.0/images/"
+		offset := len(prefix)
+		if len(urln) < offset+1 {
+			continue
+		}
+		n := urln[offset:]
+		if len(n) < inLen {
+			continue
+		}
+		if n[:inLen] == inName {
+			return n
```

Comment:
On Mon, Mar 09, 2015 at 01:22:35PM -0700, Tycho Andersen wrote:
> > +	if err != nil {
> > +		return ""
> > +	}
> > +	inLen := len(inName)
> > +	for _, urln := range imageList {
> > +		prefix := "/1.0/images/"
> > +		offset := len(prefix)
> > +		if len(urln) < offset+1 {
> > +			continue
> > +		}
> > +		n := urln[offset:]
> > +		if len(n) < inLen {
> > +			continue
> > +		}
> > +		if n[:inLen] == inName {
> > +			return n
> 
> Can we just do a sort of the list and take the shortest one here, so we don't need all this code?

The fingerprints (which we are ranging over) are supposed to all
be same size here.  The rest of the code is just for safety.
I don't see what sorting would buy us.

There may be something clever we can do with sorting into buckets,
but I don't think it would simplify the code.

I'll write a common helper for getting the object from the url
which will clean it up a bit.